smallfile-wrapper/smallfile-wrapper.py

```diff
@@ -0,0 +1,119 @@
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import elasticsearch
+import os
+import sys
+import json
+import traceback
+
+
+# The purpose of this function is to consolidate all result jsons in to 1 list
+def _load_result_JSON(smallfile_path, operations):
+    results = []
+    for operation in operations:
+        # smallfile results are stored in directory, {{smallfile_path}}/RESULTS
+        # naming convention for result JSON is: SMALLFILE_$operation_RUN_RESULT
+        _path = smallfile_path+"/RESULTS/SMALLFILE_"+operation+"_RUN_RESULT"
+        with open(_path) as json_file:
+            data = json.load(json_file)
+            results.append(data)
+    return results
+
+
+# The purpose of this function is to establish connection with
+# elasticsearch server and index the consolidated results 1 by 1
+def _index_result(server, port, payload):
+    index = "smallfile-results"
+    try:
+        es = elasticsearch.Elasticsearch([
+            {'host': server, 'port': port}], send_get_body_as='POST')
+        for result in payload:
+            es.index(index=index, doc_type="result", body=result)
+    except Exception as e:
+        sys.stdout.write("******Exception Occured*******")
+        sys.stdout.write(str(e))
+        sys.stdout.write(traceback.format_exc())
+        return False
+
+
+# The pupose of this function is to inject the uuid and user into
+# the data generated by smallfile.
+def _json_payload(data, uuid, user):
+    processed = []
+    for item in data:
+        item["workload"] = "smallfile"
+        item["uuid"] = uuid
+        item["user"] = user
+        processed.append(item)
+    return processed
+
+
+def main():
+
+    # retrieving elasticsearch server address and operating port from
+    # environement variables. The credentials are exported in the
+    # tasks of ripsaw/roles/smallfile-bench. These variables are exported
+    # based on condition, if the es_port and es_server is defined in
+    # smallfile CRDS of ripsaw. Use of "os.environ.get($env_var)" is to look
+    # for the supplied $env_var. If there is no such $env_var defined,
+    # then it will return None object.
+    server = os.environ.get("es")
+    # Use of "os.environ.get($env_var)" is to look
+    # for the supplied $env_var. If there is no such $env_var defined,
+    # then it will return None object.
+    port = os.environ.get("es_port")
+    # for each benchmark run, ripsaw generates a UUID and export it
+    # as a environement variable. Here, that uuid and a test_user is
+    # is trying to be fetched from env. Use of "os.environ.get($env_var)"
+    # is to look for the supplied $env_var. If there is no such $env_var
+    # defined, then it will return None object.
+    uuid = os.environ.get("uuid")
+    user = os.environ.get("test_user")
+
+    # Smallfile results path is based on the run, if it is executed on
```

Comment:
lets use argparse so the user running this wrapper knows the inputs without having to read code.

Reply:
sounds like a good idea. Will add it.

Reply:
Got a condition here, what sort of parameters we are looking for , that user can provide at the cli ? UUID, USER, ES_SERVER, PORT are captured through environment variables